Sources/GRPC/GRPCClientConnection.swift

```diff
@@ -171,7 +153,163 @@ open class GRPCClientConnection {
   }
 }
 
+// MARK:- Configuration structures
+
+/// A target to connect to.
+public enum ConnectionTarget {
+  /// The host and port.
+  case hostAndPort(String, Int)
+  /// The path of a Unix domain socket.
+  case unixDomainSocket(String)
+  /// A NIO socket address.
+  case socketAddress(SocketAddress)
+
+  var host: String? {
+    guard case .hostAndPort(let host, _) = self else {
+      return nil
+    }
+    return host
+  }
+}
+
 extension GRPCClientConnection {
+  /// The configuration for a connection.
+  public struct Configuration {
+    /// The target to connect to.
+    public var target: ConnectionTarget
```

Comment:
Should all these be `var` or `let`? (The struct itself is probably stored as `let`, anyway, but still.)

Reply:
`var`, since sometimes it's easier to conditionally set configuration after the fact (I think it's used somewhere in this PR to set TLS configuration).

More generally though I'd say if you want a struct to be immutable you should store it in a `let` as opposed to forcing it on the property level.